src/lib/krb5/krb/princ_comp.c

```diff
@@ -36,6 +36,10 @@ realm_compare_flags(krb5_context context,
     const krb5_data *realm1 = &princ1->realm;
     const krb5_data *realm2 = &princ2->realm;
 
+    if (princ1 == NULL || princ2 == NULL)
+        return FALSE;
+    if (realm1 == NULL || realm2 == NULL)
```

Comment:
Probably moot, but realm1 and realm2 cannot be null, as they are assigned the addresses of structure fields just above.

Reply:
Wouldn't the `princ1` / `princ2` checks on line 39 also be moot, since we've already dereference them on line 36/37 (and, would have presumably crashed if we did dereference a `NULL` pointer there)? I'm not sure if the behavior is modified by `realm1` / `realm2` being declared `const` though.

Reply:
No, simply computing &princ1->realm and &princ2->realm, doesn't dereference the princ1 and princ2 pointers; it just adds the offset to them.  (At least in practice; at a language definition level I'm not certain.)  The const qualifier is irrelevant.

Reply:
Ah, thanks for the explanation. :)

(Still, kinda seems weird to refer to the address of something in the struct before checking that the pointer to the struct isn't NULL, but I guess that's a side effect of enforcing C89's variables-before-code).

Reply:
I think at the language definition level computing the pointer is still undefined behavior.  Consider the case of a segmented architecture, where you end up trying to load in a segment that doesn't exist.